db/dcrpg/pgblockchain.go

```diff
@@ -1826,10 +1885,15 @@ func (pgb *ChainDB) StoreBlock(msgBlock *wire.MsgBlock, winningTickets []string,
 	// Convert the wire.MsgBlock to a dbtypes.Block
 	dbBlock := dbtypes.MsgBlockToDBBlock(msgBlock, pgb.chainParams)
 
-	// Get the previous winners (stake DB pool info cache have this info)
+	// Get the previous winners (stake DB pool info cache have this info). If
+	// the previous block is side chain, stakedb will not have the
+	// winners/validators. Since Validators are only used to identify misses in
+	// InsertVotes, we will leave the Validators empty and assume there are no
+	// misses. TODO: ensure this wouldn't break reorg if this block becomes main
+	// chain at some point.
```

Comment:
If a block that is in a side chain can be brought back as a mainchain block what is the likelihood of this index being violated? 
```golang
IndexBlockTableOnHash = `CREATE UNIQUE INDEX uix_block_hash ON blocks(hash);`
```

Reply:
Being "brought back" does not involve the hash changing, so it would be violated if we tried to re-insert it, but we update the row with that hash so it won't be violated in a reorg.

The issue I was worried about is that there is no ticket pool info data ( for blocks imported this way (as side chain to start), and I was not certain that would be an issue if such a block somehow became mainchain via a reorg.

So, talking this through, the primary "pool info" not available to imported side chain blocks are the winning tickets (that vote in the next block) and the validators (the winning tickets that should be spent on votes in the block).  The validators are used in InsertVotes to identify missed votes as validators that did not vote.  Imported side chain blocks will just have no misses recorded in the misses table.  But if that block became main chain, the reorganization process will actually connect these blocks in stakedb, thus providing this information when `dcrpg.(*ChainMonitor).switchToSideChain` calls `(*ChainDB).StoreBlock` again, this time with `isMainchain=true`.  This time, `InsertVotes` will have the validators and will thus be able to insert any new misses.

To summarize, a reorganization will involve connecting the blocks in stakedb, allowing the validators to be known, and all the inserts will be run again, including the misses that were not known while the block was an imported side chain block.